diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -58,6 +58,69 @@ int DifferentialFunctionComparator::cmpGEPs(const GEPOperator *GEPL,
         // The original function says the GEPs are equal - return the value
         return OriginalResult;
 
+    if (isa<ArrayType>(GEPL->getSourceElementType())
+        && isa<ArrayType>(GEPR->getSourceElementType())) {
+        // The number of elements of two enums can be different and
+        // its not considered as semantic change
+        if (GEPL->getSourceElementType()->getArrayNumElements()
+            == GEPR->getSourceElementType()->getArrayNumElements())
+            return OriginalResult;
+
+        auto *STyL = cast<SequentialType>(GEPL->getSourceElementType());
+        auto *STyR = cast<SequentialType>(GEPR->getSourceElementType());
+        if (int Res = cmpTypes(STyL->getElementType(), STyR->getElementType()))
+            return Res;
+
+        if (int Res =
+                    cmpNumbers(GEPL->getNumOperands(), GEPR->getNumOperands()))
+            return Res;
+
+        for (unsigned i = 0, e = GEPL->getNumOperands(); i != e; ++i) {
+            if (int Res = cmpValues(GEPL->getOperand(i), GEPR->getOperand(i)))
+                return Res;
+        }
+
+        if (GEPL->getNumOperands() < 3 || GEPR->getNumOperands() < 3)
+            return OriginalResult;
```

Comment:
```suggestion
        if (GEPL->getNumOperands() != 3 || GEPR->getNumOperands() != 3)
            return OriginalResult;
```
Add a comment (e.g. "We only handle GEPs that access exactly one array element") and move this upwards to the beginning of this block.